### ISSUE TITLE
bug: Brier score incorrectly available for multiclass classification problems

### ISSUE THREAD

**Issue by Muhammad-Rebaal** [probabl-ai/skore#1637]
### Describe the bug

The `EstimatorReport` class incorrectly exposes the Brier score metric for multiclass classification problems, even though this metric is only mathematically valid for binary classification. This could lead users to calculate and rely on invalid metrics for multiclass models.

### Steps/Code to Reproduce

```
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from skore import EstimatorReport

# Create a multiclass classification dataset (3 classes)
X, y = make_classification(n_classes=3, n_clusters_per_class=1, random_state=42)
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

# Create and fit a LogisticRegression model
estimator = LogisticRegression()
estimator.fit(X_train, y_train)

# Create an EstimatorReport
report = EstimatorReport(estimator, X_test=X_test, y_test=y_test)

# Bug: This should return False for multiclass problems, but returns True
print(hasattr(report.metrics, "brier_score"))
```

### Expected Behavior

For multiclass classification problems (where `y` contains more than 2 unique classes), the `report.metrics, "brier_score"` should return False, as the Brier score is only defined for binary classification problems.

### Actual Behavior

When running with a multiclass dataset, `report.metrics, "brier_score"` returns True and allows users to calculate a Brier score for multiclass problems, which is mathematically invalid:
```
True
Brier score (invalid for multiclass): [invalid score value]
```
This contradicts the existing test case test_estimator_report_brier_score_requires_binary_classification which specifically checks that this attribute should not be available for multiclass problems.

### Environment

```shell
System:
    python: 3.12.6 (tags/v3.12.6:a4a2d2b, Sep  6 2024, 20:11:23) [MSC v.1940 64 bit (AMD64)]
executable: C:\Program Files\Python312\python.exe
   machine: Windows-11-10.0.22000-SP0

Python dependencies:
        skore: 0.8.2
          pip: 25.1
    diskcache: 5.6.3
       joblib: 1.4.2
   matplotlib: 3.10.0
        numpy: 1.26.4
       pandas: 2.2.3
         rich: 14.0.0
 scikit-learn: 1.6.1
        skops: 0.11.0
```

**Comment by Muhammad-Rebaal** [comment]
Hi @sylvaincom , @auguste-probabl !

Could you take a look at this issue and assign this to me I found out when testing the `test_estimator.py` and accidently pushed in another issue and then discarded as per your request.

**Comment by glemaitre** [comment]
I solved this bug some times ago and it has been merged in `main`.

The reproducer above provides:

![Image](https://github.com/user-attachments/assets/253c349b-d91a-4b54-8249-964b9e81682a)

I know that we still have a choice to do when comparing report but for the estimator and cross-validation reports, we should be fine with non-regression tests included in the code base.
